wallet/db.c

```diff
@@ -1697,7 +1697,10 @@ void db_bind_preimage(struct db_stmt *stmt, int pos, const struct preimage *p)
 
 void db_bind_sha256(struct db_stmt *stmt, int pos, const struct sha256 *s)
 {
-	db_bind_blob(stmt, pos, s->u.u8, sizeof(struct sha256));
+	if(!s)
+		db_bind_null(stmt, pos);
+	else
+		db_bind_blob(stmt, pos, s->u.u8, sizeof(struct sha256));
```

Comment:
NACK, the DB API is very simple on purpose and should not decide the nullability of a field on its own. That's for the `wallet.c` to disentangle where we actually have more information about the schema we're dealing with. This could potentially introduce some really subtle bugs.

Reply:
I see, and I agree with your design choice. this check usually is also scary in the API.

Thanks 😸